index.js

```diff
@@ -33,6 +33,18 @@ function normalize(type, value) {
   return result;
 }
 
+function conforms(type, value) {
```

Comment:
is it best to support all 3 types (string, array, function) or should it just support function and the utility methods like `string()`, `boolean()`, etc can just have the predicates applied?